core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -706,20 +843,24 @@ class LogManager(logDirs: Array[File],
   /**
    * Get all the partition logs
    */
-  def allLogs(): Iterable[Log] = logs.values
+  def allLogs(): Iterable[Log] = logs.values ++ futureLogs.values
 
   /**
    * Get a map of TopicPartition => Log
    */
   def logsByTopicPartition: Map[TopicPartition, Log] = logs.toMap
 
+  def logsByTopic(topic: String): Seq[Log] = {
```

Comment:
This method is only used in test. We can not naively include futureLogs in logsByTopicPartition because it returns a map with TopicPartition as key. I have updated the patch to remove this method and replaced its use in test with LogManger.getLog().